plugins/modules/ndo_l3out_node_group_policy.py

```diff
@@ -0,0 +1,316 @@
+        target_dscp=dict(type="str", choices=list(TARGET_DSCP_MAP)),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name"]],
+            ["state", "present", ["name"]],
+            ["bfd_multi_hop_authentication", "enabled", ["bfd_multi_hop_key_id", "bfd_multi_hop_key"]],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    l3out = module.params.get("l3out")
+    name = module.params.get("name")
+    description = module.params.get("description")
+    node_routing_policy = module.params.get("node_routing_policy")
+    bfd_multi_hop_authentication = module.params.get("bfd_multi_hop_authentication")
+    bfd_multi_hop_key_id = module.params.get("bfd_multi_hop_key_id")
+    bfd_multi_hop_key = module.params.get("bfd_multi_hop_key")
+    target_dscp = TARGET_DSCP_MAP.get(module.params.get("target_dscp"))
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "l3out", template)
+    mso_template.validate_template("l3out")
+
+    l3out_object = mso_template.get_l3out_object(name=l3out, fail_module=True)
+    l3out_node_group = mso_template.get_l3out_node_group(name, l3out_object.details)
+
+    if name:
+        if l3out_node_group:
+            mso.existing = mso.previous = copy.deepcopy(l3out_node_group.details)  # Query a specific object
+    elif l3out_node_group:
+        mso.existing = l3out_node_group  # Query all objects
+
+    if state != "query":
+        node_group_policy_path = "/l3outTemplate/l3outs/{0}/nodeGroups/{1}".format(l3out_object.index, l3out_node_group.index if l3out_node_group else "-")
+
+    ops = []
+
+    if state == "present":
+        l3out_node_routing_policy_object = None
+        if node_routing_policy:
+            l3out_node_routing_policy_objects = mso.query_objs(
+                generate_api_endpoint(
+                    "templates/objects", **{"type": "l3OutNodePolGroup", "tenant-id": mso_template.template_summary.get("tenantId"), "include-common": "true"}
+                )
+            )
+            l3out_node_routing_policy_object = mso_template.get_object_by_key_value_pairs(
+                "L3Out Node Routing Policy", l3out_node_routing_policy_objects, [KVPair("name", node_routing_policy)], True
+            )
+
+        if mso.existing:
+            proposed_payload = copy.deepcopy(mso.existing)
+
+            if description is not None and mso.existing.get("description") != description:
+                ops.append(dict(op="replace", path=node_group_policy_path + "/description", value=description))
+                proposed_payload["description"] = description
+
+            if (
+                node_routing_policy is not None
+                and l3out_node_routing_policy_object
```

Comment:
can't this be simplified just checking l3out_node_routing_policy_object? because that should only have a value when node_routing_policy is not None?